demos/gallery/samples/common/HTMLWidget.js

```diff
@@ -0,0 +1,58 @@
+import { HTMLWidget } from "@hpcc-js/common";
+
+const data = [
+    ["A", 100],
+    ["B", 162],
+    ["C", 262],
+];
+class HTMLExample extends HTMLWidget{
+    constructor(){
+        super();
+    }
+    enter(domNode, element){
+        super.enter.apply(this, arguments);
+        this._div = element.append("div");
+    }
+    update(domNode, element){
+        super.update.apply(this, arguments);
+        const gutter = 4;
+        const rowHeight = 30;
+        const labelWidth = 10;
+        
+        const rowsEnter = this._div.selectAll(".data-row").data(this.data())
+            .enter()
+            .append("div")
+            .style("width","100%")
+            .style("height",rowHeight+"px")
+            .style("margin-top",gutter+"px")
+            .classed("data-row", true)
+            ;
+        rowsEnter.append("span");
+        rowsEnter.append("div");
+        rowsEnter
+            .merge(rowsEnter)
+            .style("color","red")
+            .style("float","left")
+            .each(function(row,i){
+                const text = this.children[0];
+                text.style.float = "left";
+                text.style.display = "inline-block";
+                text.style.width = labelWidth+"px";
+                text.innerText = row[0];
+                const rect = this.children[1];
+                rect.style.float = "left";
+                text.style.display = "inline-block";
+                rect.style.marginLeft = gutter+"px";
+                rect.style.backgroundColor = "red";
+                rect.style.width = row[1]+"px";
+                rect.style.height = rowHeight+"px";
+            })
+            .attr("transform",(row,i)=>{
+                const x = gutter;
+                const y = (rowHeight * i) + (gutter * (i+1)) + rowHeight;
+                return `translate(${x} ${y})`;
+            })
+            ;
```

Comment:
Can you clean up the `d3` code so its more "standard d3".

Missing `exit` call

Reply:
@jbrundage bump

Reply:
Missing exit call.